core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -470,6 +470,29 @@ private[spark] object PythonRDD extends Logging {
     SocketAuthHelper.serveToStream(threadName, authHelper)(writeFunc)
   }
 
+  /**
+   * Create a socket server object and background thread to execute the writeFunc
+   * with the given OutputStream.
+   *
+   * This is the same as serveToStream, only it returns a server object that
+   * can be used to sync in Python.
+   */
+  private[spark] def serveToStreamWithSync(
```

Comment:
This could be cleaned up and replace the existing `serveToStream`. It just returns the `SocketAuthServer` object as the third element in the Array, and it could be ignored if no synchronization is needed.